feature-proxy-api/src/main/java/io/novafoundation/nova/feature_proxy_api/data/common/ProxyDepositCalculator.kt

```diff
@@ -0,0 +1,16 @@
+package io.novafoundation.nova.feature_proxy_api.data.common
+
+import io.novafoundation.nova.runtime.multiNetwork.chain.model.Chain
+import java.math.BigInteger
+
+class DepositBaseAndFactor(
+    val baseAmount: BigInteger,
+    val factorAmount: BigInteger
+)
+
+interface ProxyDepositCalculator {
+
+    suspend fun getDepositConstants(chain: Chain): DepositBaseAndFactor
+
+    fun calculateProxyDeposit(baseAndFactor: DepositBaseAndFactor, proxiesCount: Int): BigInteger
```

Comment:
Why do we need to separate thos two methods? From the usages it looks like we only use them together.
Imho, itis better to either 
1. Only leave `calculateProxyDeposit` here OR
2. Leave `getDepositConstants` here and make `calculateProxyDeposit` and extension function for `DepositBaseAndFactor DepositBaseAndFactor`